pkg/operator/deploy/deploy.go

```diff
@@ -212,6 +212,13 @@ func (o *operator) resources() ([]kruntime.Object, error) {
 		serviceSubnets = append(serviceSubnets, "/subscriptions/"+o.env.SubscriptionID()+"/resourceGroups/"+o.env.GatewayResourceGroup()+"/providers/Microsoft.Network/virtualNetworks/gateway-vnet/subnets/gateway-subnet")
 	}
 
+	// this service subnet is added during install time if cluster is installed via Hive
+	if o.oc.Properties.HiveProfile.CreatedByHive {
+		// TODO: add AKS shard support
+		hiveShard := 1
+		serviceSubnets = append(serviceSubnets, fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/aks-net/subnets/PodSubnet-%03d", o.env.SubscriptionID(), o.env.ResourceGroup(), hiveShard))
+	}
```

Comment:
This change feels out-of-scope for this work, was it included by mistake or is it somehow reducing API calls?

Reply:
tl;dr I'll remove this, see below. 


This was done intentionally, but not fully correctly, to keep our two separate Storage Account reconciliation processes, RP-side (cluster creation+PUCM) vs operator, in sync with what they put down. The RP-side reconciliation will include this subnet NACL on just the cluster storageaccount for Hive-created clusters, but the operator would not. 

After re-reading this, it appears that this change as-is will include this NACL on both storageaccounts during operator reconciliation, in reality actually creating the conflict that this change was trying to mitigate (dueling definitions between PUCM and operator).